### PR TITLE
s390x enablement WIP

### LINKED ISSUE #3356: s390x enablement
**Is your feature request related to a problem? Please describe**:
port of kubevirt cdi to s390x architecture

**Describe the solution you'd like**:
kubevirt cdi builds on and for s390x architecture

**Describe alternatives you've considered**:

**Additional context**:

### PR DESCRIPTION
**Pull request by cfilleke**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3356 

**Special notes for your reviewer**:
WIP still needs rebase/merge conflict resolution

**Release note**:

```release-note
enables s390x native build
```

### CODE REVIEW

**Comment by awels** [comment]
This is the one we want in, together with the fix for the unit test? I can waive the failing unit tests for this PR, so we can get this in so the other PR can pass. Although I think the other PR would have no issue merging even without this.

**Comment by cfilleke** [comment]
If we're going to be using cross-compile with this, I'll need to merge the changes to the Dockerfile first; however, if we can build on the new cloud s390x external ci/cd cluster (and I would suggest we do that) then we need to configure that to build & register the build-the-builder container first; also have some questions about at what stage in the build-test lifecycle the `make rpm-deps` is executed.  Thanks!

**Comment by dominikholler** [comment]
Building containerized-data-importer for s390x failed for me with 

```
 final command line arguments: "build" "-o" "/remote-source/app/_out/cmd/cdi-importer/cdi-importer" "-tags" "strictfipsruntime" "-ldflags" "-extldflags $static_flag" "-ldflags" "-X 'kubevirt.io/containerized-data-importer/pkg/version.buildDate=2024-08-15T15:58:56Z'"
invoking real go binary
 # kubevirt.io/containerized-data-importer/pkg/util
 ../../pkg/util/util.go:116:9: invalid operation: int64(stat.Bavail) * stat.Bsize (mismatched types int64 and uint32)
Exited with: 1
```

**Comment by awels** [comment]
Can you rebase? Seems like there are some issues there now.

**Comment by awels** [comment]
To fix, run `make generate` and `make generate-verify` to ensure nothing is being generated

File: pkg/importer/BUILD.bazel [hunk 1]
```diff
@@ -93,7 +97,6 @@ go_test(
         "//tests/utils:go_default_library",
         "//vendor/cloud.google.com/go/storage:go_default_library",
         "//vendor/github.com/aws/aws-sdk-go/service/s3:go_default_library",
-        "//vendor/github.com/containers/image/v5/types:go_default_library",
```
Comment (awels) [inline pkg/importer/BUILD.bazel:96]: This is the line the verify is complaining about. It wants to add it back again. I take it this is not available in s390x?

Comment (cfilleke) [reply pkg/importer/BUILD.bazel]: When I ran `git difftool -y --extcmd=hack/diff-csv.sh` manually it found an issue with the indentation in 2 lines in `tools/cdi-func-test-registry-init/BUILD.bazel` as well.

**Comment by cfilleke** [comment]
`pull-cdi-unit-test` fails with the same error that updating `rpm/README.md` in PR https://github.com/kubevirt/containerized-data-importer/pull/3409 fails, so it's not clear that this would be resolved with a code change @awels

**Comment by cfilleke** [comment]
All tests passed except 'tide' and this apears to only require only /lgtm /approve now @aglitke @alromeros @ShellyKa13 but it's still an XXL so please feel free to ask me questions. Thanks!

**Comment by cfilleke** [comment]
`make all` build test on s390x results here: https://gist.github.com/cfilleke/a4dcbad9a9ce9472ad3d74bdbedd4d62

**Review by awels — APPROVED** [review]
/approve

**Pull request merged**